File: haushaltsbuch/src/main/java/org/github/haschi/haushaltsbuch/api/WährungsbetragSerialisierer.java
package org.github.haschi.haushaltsbuch.api;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.ser.std.StdSerializer;

import java.io.IOException;

public class WährungsbetragSerialisierer extends StdSerializer<Währungsbetrag>
{
    public WährungsbetragSerialisierer() {
        this(null);
    }

    public WährungsbetragSerialisierer(final Class<Währungsbetrag> t) {
        super(t);
    }

    @Override
    public void serialize(
            final Währungsbetrag value,
            final JsonGenerator jgen,
            final SerializerProvider provider)
        throws IOException
    {
        jgen.writeString(value.toString());
    }
}

File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/api/WährungsbetragSerialisierer.kt
package org.github.haschi.haushaltsbuch.api

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.ser.std.StdSerializer

import java.io.IOException

class WährungsbetragSerialisierer @JvmOverloads constructor(t: Class<Währungsbetrag>? = null) : StdSerializer<Währungsbetrag>(t) {

    @Throws(IOException::class)
    override fun serialize(
            value: Währungsbetrag,
            jgen: JsonGenerator,
            provider: SerializerProvider) {
        jgen.writeString(value.toString())
    }
}
